ui/src/adapters/api/credentials.ts

```diff
@@ -87,12 +161,91 @@ export const credentialStatusParser = getStrictParser<CredentialStatus>()(
 export async function getCredential({
   credentialID,
   env,
+  issuerIdentifier,
   signal,
 }: {
   credentialID: string;
   env: Env;
+  issuerIdentifier: IssuerIdentifier;
   signal?: AbortSignal;
 }): Promise<Response<Credential>> {
+  try {
+    const credentialDetail = await getCredentialDetail({
+      credentialID,
+      env,
+      issuerIdentifier,
+      signal,
+    });
+
+    if (!credentialDetail.success) {
+      return buildErrorResponse(credentialDetail.error);
+    }
+
+    const {
+      "@context": context,
+      credentialSchema,
+      credentialStatus,
+      credentialSubject,
+      expirationDate,
+      id,
+      issuanceDate,
+      issuer,
+      proof,
+    } = credentialDetail.data;
+
+    const revocationStatus = await getRevocationStatus({
+      env,
+      issuerIdentifier,
+      params: { nonce: credentialStatus.revocationNonce },
+      signal,
+    });
+
+    if (!revocationStatus.success) {
+      return buildErrorResponse(revocationStatus.error);
+    }
+
+    const proofTypes = proof.map(({ type }) => type);
+
+    const schemaHash = getSchemaHash({
+      id: `${context.at(-1)}#${credentialSubject.type}`,
+      name: credentialSubject.type,
+    });
+
+    const revoked = revocationStatus.data.mtp.existence;
+    const expired = expirationDate ? new Date() > new Date(expirationDate) : false;
+
+    return buildSuccessResponse(
+      credentialParser.parse({
+        createdAt: issuanceDate,
+        credentialSubject,
+        expired,
+        expiresAt: expirationDate,
+        id: id.split(":").at(-1),
+        proofTypes,
+        revNonce: credentialStatus.revocationNonce,
+        revoked,
+        schemaHash: schemaHash.success ? schemaHash.data : "",
```

Comment:
?

Reply:
the getSchemaHash function can return the success: false

Reply:
Because it can error. When this happens, we need to propagate the error, not silently swallow it like this.